pkgs/development/python-modules/pybind11/default.nix

```
@@ -34,9 +34,9 @@ buildPythonPackage rec {
     "-DBoost_INCLUDE_DIR=${lib.getDev boost}/include"
     "-DEIGEN3_INCLUDE_DIR=${lib.getDev eigen}/include/eigen3"
     "-DBUILD_TESTING=on"
-  ] ++ lib.optionals (python.isPy3k && !stdenv.cc.isClang) [
-    "-DPYBIND11_CXX_STANDARD=-std=c++17"
-  ];
+    "-DPYTHON_EXECUTABLE:FILEPATH=${python.pythonForBuild.interpreter}"
+  ] ++ lib.optional (python.isPy3k && !stdenv.cc.isClang)
+       "-DPYBIND11_CXX_STANDARD=-std=c++17";
```

please keep using `optionals` here:
```suggestion
  ] ++ lib.optionals (python.isPy3k && !stdenv.cc.isClang) [
    "-DPYBIND11_CXX_STANDARD=-std=c++17"
  ]
```
Besides that, LGTM.

---

done (didn't use your suggestion because missing `;` in line 40).

Checked:
- `nix build  .#python39Packages.pybind11` (no change)
- `nix build  .#python38Packages.pybind11` (also ok)
- `nix build .#pkgsCross.aarch64-multiplatform.python39Packages.pybind11` (no change)
- `nix build .#pkgsCross.aarch64-multiplatform.python38Packages.pybind11` (also ok)